### ISSUE TITLE
import from URL

### ISSUE THREAD

**Issue by genmeblog** [techascent/tech.ml.dataset#33]
Could you consider an option to add reading data from URL?

It would be nice to have:

```clojure
(ds/->dataset "https://raw.githubusercontent.com/Rdatatable/data.table/master/vignettes/flights14.csv")
```

Which (on beta-22) throws ` Missing config value: :tech-io-cache-local` exception.

Workaround:

```clojure
(ds/->dataset (.openStream (java.net.URL. "https://raw.githubusercontent.com/Rdatatable/data.table/master/vignettes/flights14.csv")))
```

**Comment by keesterbrugge** [comment]
Hi @genmeblog,

that code works for me. I'm on version 2.0-beta-21

**Comment by harold** [comment]
The first one is a great idea, and already works:

```(clojure)
user> (require '[tech.ml.dataset :as ds])
user> (ds/->dataset "https://raw.githubusercontent.com/Rdatatable/data.table/master/vignettes/flights14.csv")

#<tech.ml.dataset.impl.dataset.Dataset@51098478 https://raw.githubusercontent.com/Rdatatable/data.table/master/vignettes/flights14.csv [253316 11]:

| year | month | day | dep_delay | arr_delay | carrier | origin | dest | air_time | distance | hour |
|------+-------+-----+-----------+-----------+---------+--------+------+----------+----------+------|
| 2014 |     1 |   1 |        14 |        13 |      AA |    JFK |  LAX |      359 |     2475 |    9 |
| 2014 |     1 |   1 |        -3 |        13 |      AA |    JFK |  LAX |      363 |     2475 |   11 |
| 2014 |     1 |   1 |         2 |         9 |      AA |    JFK |  LAX |      351 |     2475 |   19 |
| 2014 |     1 |   1 |        -8 |       -26 |      AA |    LGA |  PBI |      157 |     1035 |    7 |
| 2014 |     1 |   1 |         2 |         1 |      AA |    JFK |  LAX |      350 |     2475 |   13 |
| 2014 |     1 |   1 |         4 |         0 |      AA |    EWR |  LAX |      339 |     2454 |   18 |
| 2014 |     1 |   1 |        -2 |       -18 |      AA |    JFK |  LAX |      338 |     2475 |   21 |
| 2014 |     1 |   1 |        -3 |       -14 |      AA |    JFK |  LAX |      356 |     2475 |   15 |
| 2014 |     1 |   1 |        -1 |       -17 |      AA |    JFK |  MIA |      161 |     1089 |   15 |
| 2014 |     1 |   1 |        -2 |       -14 |      AA |    JFK |  SEA |      349 |     2422 |   18 |
| 2014 |     1 |   1 |        -5 |       -17 |      AA |    EWR |  MIA |      161 |     1085 |   16 |
| 2014 |     1 |   1 |         7 |        -5 |      AA |    JFK |  SFO |      365 |     2586 |   17 |
| 2014 |     1 |   1 |         3 |         1 |      AA |    JFK |  BOS |       39 |      187 |   12 |
| 2014 |     1 |   1 |       142 |       133 |      AA |    JFK |  LAX |      345 |     2475 |   19 |
| 2014 |     1 |   1 |        -5 |       -26 |      AA |    JFK |  BOS |       35 |      187 |   17 |
| 2014 |     1 |   1 |        18 |        69 |      AA |    JFK |  ORD |      155 |      740 |   17 |
| 2014 |     1 |   1 |        25 |        36 |      AA |    JFK |  IAH |      234 |     1417 |   16 |
| 2014 |     1 |   1 |        -1 |         1 |      AA |    JFK |  AUS |      232 |     1521 |   17 |
| 2014 |     1 |   1 |       191 |       185 |      AA |    EWR |  DFW |      214 |     1372 |   16 |
| 2014 |     1 |   1 |        -7 |        -6 |      AA |    LGA |  ORD |      142 |      733 |    5 |
| 2014 |     1 |   1 |        -7 |         0 |      AA |    LGA |  ORD |      143 |      733 |    6 |
| 2014 |     1 |   1 |        -8 |       -17 |      AA |    LGA |  ORD |      139 |      733 |    6 |
| 2014 |     1 |   1 |        -2 |        15 |      AA |    LGA |  ORD |      145 |      733 |    7 |
| 2014 |     1 |   1 |        -3 |         1 |      AA |    LGA |  ORD |      139 |      733 |    8 |
| 2014 |     1 |   1 |        44 |        42 |      AA |    LGA |  ORD |      141 |      733 |   10 |
```

**Comment by genmeblog** [comment]
On 2.0-beta-22 I'm getting an exception:

```clojure
1. Unhandled java.lang.IllegalArgumentException
   Missing config value: :tech-io-cache-local

                  core.clj:  198  tech.config.core/get-config
                  core.clj:  193  tech.config.core/get-config
             providers.clj:   51  tech.io.providers/fn/fn
                  core.clj: 2753  clojure.core/map/fn
              LazySeq.java:   42  clojure.lang.LazySeq/sval
              LazySeq.java:   51  clojure.lang.LazySeq/seq
                   RT.java:  535  clojure.lang.RT/seq
                  core.clj:  137  clojure.core/seq
                  core.clj: 2809  clojure.core/filter/fn
              LazySeq.java:   42  clojure.lang.LazySeq/sval
              LazySeq.java:   51  clojure.lang.LazySeq/seq
                   RT.java:  535  clojure.lang.RT/seq
                  core.clj:  137  clojure.core/seq
                  core.clj:  930  clojure.core/reduce1
                  core.clj:  947  clojure.core/reverse
                  core.clj:  947  clojure.core/reverse
             providers.clj:   42  tech.io.providers/provider-seq->wrapped-providers
             providers.clj:   35  tech.io.providers/provider-seq->wrapped-providers
             providers.clj:   48  tech.io.providers/fn
             providers.clj:   47  tech.io.providers/fn
            ...
```

**Comment by genmeblog** [comment]
the same behaviour under `2.0-beta-21`. I'm on OpenJDK 11.

**Comment by genmeblog** [comment]
Ok. Looks like something must be wrong with my dependencies.

**Comment by genmeblog** [comment]
The problem appears when I add `cider-nrepl` as a plugin.

```clojure
  :plugins [[cider/cider-nrepl "0.25.0-alpha1"]]
  :dependencies [[org.clojure/clojure "1.10.1"]
                 [techascent/tech.ml.dataset "2.0-beta-22"]]
```

**Comment by harold** [comment]
That is pretty odd.

@cnuernber - it looks like maybe this: https://github.com/techascent/tech.io/blob/17a3cf30fcfe88d83a023ea0586c8ddca3a4bd4e/src/tech/io/providers.clj#L51
 - Perhaps a try/catch there, as get-config can throw.

@genmeblog - you may be able to work around this with something like:

```(clojure)
(tech.config.core/set-config! :tech-io-cache-local false)
(tech.config.core/set-config! :tech-io-redirect-local false)
```

Though, it's unclear why including the nrepl plugin would have any effect in the first place.

**Comment by genmeblog** [comment]
`set-config!` indeed helps. Why cider-nrepl interfer? I have no idea. Especially when it's loaded as plugin.

**Comment by cnuernber** [comment]
try/cache for sure seems to me the right way.  Odd we have never hit that but who knows.

**Comment by harold** [comment]
> Why cider-nrepl interfer? I have no idea.
>

Agreed - shouldn't affect anything. I tried adding that plugin to my project and I can still load the dataset as normal.

We have encountered similar issues in the past with config values not flowing. The system depends on dependency configuration files being hierarchically included through `resources/`. There are a variety of ways that can break down.

Another, more permanent fix would be to add a file called `app-config.edn` to your resources with contents like:

```
{:tech-io-cache-local false
 :tech-io-redirect-local false}
```

Which will save you from needing to use `set-config!`
